experimental/packages/opentelemetry-sdk-node/src/TracerProviderWithEnvExporter.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag } from '@opentelemetry/api';
+import { getEnv, getEnvWithoutDefaults } from '@opentelemetry/core';
+import { ConsoleSpanExporter, SpanExporter, BatchSpanProcessor, SimpleSpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { NodeTracerConfig, NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { OTLPTraceExporter as OTLPProtoTraceExporter } from '@opentelemetry/exporter-trace-otlp-proto';
+import { OTLPTraceExporter as OTLPHttpTraceExporter} from '@opentelemetry/exporter-trace-otlp-http';
+import { OTLPTraceExporter as OTLPGrpcTraceExporter} from '@opentelemetry/exporter-trace-otlp-grpc';
+import { ZipkinExporter } from '@opentelemetry/exporter-zipkin';
+import { JaegerExporter } from '@opentelemetry/exporter-jaeger';
+
+export class TracerProviderWithEnvExporters extends NodeTracerProvider {
```

Comment:
I'm not too familiar with the trace setup code: Would it be possible for this to be a simple `EnvSpanProcessorFactory` that only creates `SpanProcessor`s with associated `TraceExporter`s based on Environment variables? :thinking:

From what I can see this `TracerProvider` implementation is never used as such, right? :thinking:

Reply:
This `TracerProviderWithEnvExporters` is used [here](https://github.com/open-telemetry/opentelemetry-js/blob/e02d73981813f0efec72642e7e3d4ee10b53d0c6/experimental/packages/opentelemetry-sdk-node/src/sdk.ts#L201-L202). 

Based on [some feedback](https://github.com/open-telemetry/opentelemetry-js/pull/3143#issuecomment-1214157894) from @legendecas and @vmarchaud, the goal was to have a class (ex. `TracerProviderWithEnvExporters`) that extends the node tracer provider with its own exporters/propagators that can then be configured by the env. This does make this feature implementation cleaner and uses some existing mechanism like the `_registeredExporters` that were created in the `BasicTracerProvider`.

Reply:
Ah, I see. I overlooked these comments, sorry. :slightly_smiling_face:
I think the change you proposed in [this comment here](https://github.com/open-telemetry/opentelemetry-js/pull/3143#discussion_r964049961) sounds like a great idea and would address my concerns. :slightly_smiling_face:.